bin/oneoff/normalize_gender

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env ruby
+#
+# This script normalizes values in the `gender` column of the `users` table to
+# be one of NULL (unspecified), "-" (decline to answer), "f" (female), or "m"
+# (male). Any existing value that is not one of these four is set to NULL.
+#
+# As of 2016-05-23, there were approximately 727886 such users in production.
+#
+# This is being done as a one-off script so as to be able to control the timing
+# of its execution in production, followed by a migration performing the same
+# task so as to assure good data in all environments.
+
+require_relative '../../dashboard/config/environment'
+require src_dir 'database'
+
+User.with_deleted.
+  where('gender IS NOT NULL').
+  where('gender != "-" AND gender != "f" AND gender != "m"').
+  update_all(gender: nil)
```

Comment:
Per documentation (http://apidock.com/rails/v4.0.2/ActiveRecord/Relation/update_all), note that this does not trigger callbacks or validations.